capsule update info (#879)
Dasharo/docs: docs/osf-trivia-list/capsule-update.md

# Frequenty Asked Questions about Capsule Update

## Q: Is fwupd still Linux only?

> A: Yes, AFAICT fwupd is Linux only. However, this isn't the only software that

Author: of course: https://github.com/Dasharo/docs/pull/879/commits/6a95132cfe6aeba5bee8dd8c84f67e2eb011bbef

uses capsules to perform the update, Windows also uses it (although this would
require creating and distributing firmware as drivers through Windows Update,
which is beyond the scope of this project). Linux can also be compiled with
EFI_CAPSULE_LOADER [1] which exposes an interface for sending capsules to the
firmware by a simple write to file. Can it really get easier than

> `cat firmware.bin > /dev/efi_capsule_loader` [2]?

Author: here: https://github.com/Dasharo/docs/pull/879/commits/da34cf01bb1c772d645ddd6c372f656e176694ad

> ## Q:Why relinquish the traditional flashrom-based method then?

Author: fixed: https://github.com/Dasharo/docs/pull/879/commits/d1343a3155fe5e08c5ba5398b43a216d317682bb

A: The flashrom's linux_mtd drivers can't handle all of the protection
mechanisms, for example SMM BWP (BIOS Write Protection) can only be
disabled by a reset (not counting bugs like broken S3 resume or misconfigured
GPIO that would allow accessing the pins directly to bit-bang raw commands).

[1] https://docs.dasharo.com/projects/capsule-updates/

> [2] https://github.com/orgs/Dasharo/projects/7/views/1

Author: already deleted them somewhere along the way: https://github.com/Dasharo/docs/pull/879/commits/5dd5b5b2370bf29fc3e4c4befe6a21b0d96f5135

# Frequenty Asked Questions about Capsule Update

## Q: Why was the traditional flashrom-based method replaced by capsule-based one?

A: There are many advantages of capsule-based update.

> Fist of all, it's much easier[1]:

Reviewer: ```suggestion
First of all, it's much easier[1]:
```

>The steps to update efi firmware are: </br>
>1.) cat firmware.cap > /dev/efi_capsule_loader </br>
>2.) reboot

Not only is it simpler but also more efficient, because there is no need for
manual reset and disabling all of the protections before update, thus the whole
process takes less time. Speaking of protection the last but not least reason is

> safety, the signed coalesced data can only be accessed by signatures, verified by
> edk2. Ensuring no unauthorized activity is allowed.

Reviewer: ```suggestion
safety, the signed coalesced data can only be accessed after signatures are verified by
edk2, ensuring no unauthorized activity is allowed.
```

verified by edk2, ensuring no unauthorized activity is allowed.

## Q: Is fwupd still Linux only?

> A: Yes, fwupd is Linux only. However, this isn't the only software that

Reviewer: ```suggestion
A: No, fwupd isn't Linux only. Even the oldest release on GitHub (Feb 3, 2020) has [fwupd-1.3.7-setup-x86_64.exe](https://github.com/fwupd/fwupd/releases/tag/1.3.7). We also added support for BSD systems as part of [fwdup-BSD](https://nlnet.nl/project/fwdup-BSD/) project.

However, fwupd isn't the only software that
```

Please also rewrap it as needed.

Author: here's the updated and squashed version: https://github.com/Dasharo/docs/pull/879/commits/e191a0608caba0d2635516a57efc0c3ab5811d8b

Reviewer 2: @wiktormowinski You can always commit suggestions using the `Commit suggestion` button, or if there are multiple, use  `Add suggestion to batch`, and apply them together. This way, the threads will be resolved automatically without needing to do it manually.

See: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

@krystian-hebel Can you please fix this part and merge the PR?

Author: @BeataZdunczyk https://github.com/Dasharo/docs/pull/879#discussion_r1721700040 that's what i did last time :c

Reviewer: You missed the empty line. Fixed now, as soon as CI finishes I'll merge this PR.

require creating and distributing firmware as drivers through Windows Update,
which is beyond the scope of this project). Linux can also be compiled with
EFI_CAPSULE_LOADER [1] which exposes an interface for sending capsules to the
firmware by a simple write to file.

> ## Q: Where in the process are signatures verified, and which parts are signed?

Reviewer: This doesn't contain answer to `which parts are signed` part. In fact, that part may be changed due to an issue reported by Sergii, so for the time being it would be safest to remove it.

```suggestion
## Q: Where in the process are signatures verified?
```

A: As mentioned above, signatures are verified by edk2, on coalesced data. There
are decisions made earlier based on presence of data, not on it's content. Hence

> the focus on not allowing booting an OS with full access to flashenabled when

Reviewer: ```suggestion
the focus on not allowing booting an OS with full access to flash enabled when
```

Reviewer 2: @wiktormowinski please do not resolve threads as an assignee. Let the reviewer decide if a thread is resolved. Also, you can add suggestions to a batch and apply them all at once: this will automatically resolve the threads on GH.

Author: oh ok

# Frequenty Asked Questions about Capsule Update

## Q: Why was the traditional flashrom-based method replaced by capsule-based one?

A: There are many advantages of capsule-based update.
First of all, it's much easier[1]:

The steps to update efi firmware are:

1. `cat firmware.cap > /dev/efi_capsule_loader`
2. `reboot`

Not only is it simpler but also more efficient, because there is no need for
manual reset and disabling all of the protections before update, thus the whole
process takes less time. Speaking of protection the last but not least reason is
safety, the signed coalesced data can only be accessed after signatures are
verified by edk2, ensuring no unauthorized activity is allowed.

## Q: Is fwupd still Linux only?

A: Yes, fwupd is Linux only. However, this isn't the only software that
uses capsules to perform the update, Windows also uses it (although this would
require creating and distributing firmware as drivers through Windows Update,
which is beyond the scope of this project). Linux can also be compiled with

> [EFI_CAPSULE_LOADER](https://lore.kernel.org/all/\

Author: commit https://github.com/Dasharo/docs/pull/879/commits/e275ff9f3e32490291c11199dece7114bc558809 addressed this problem
